go/libkb/saltpack_dec.go

```diff
@@ -4,42 +4,43 @@
 package libkb
 
 import (
-	"io"
-
 	"github.com/keybase/client/go/saltpack"
+	"io"
 )
 
 func SaltPackDecrypt(
 	source io.Reader, sink io.WriteCloser,
-	deviceEncryptionKey NaclDHKeyPair, checkSender func(*saltpack.MessageKeyInfo) error) error {
+	deviceEncryptionKey NaclDHKeyPair,
+	checkSender func(*saltpack.MessageKeyInfo) error) (*saltpack.MessageKeyInfo, error) {
+
 	mki, plainsource, frame, err := saltpack.NewDearmor62DecryptStream(
 		source, naclKeyring(deviceEncryptionKey))
 	if err != nil {
-		return err
+		return mki, err
 	}
 
 	if checkSender != nil {
 		if err = checkSender(mki); err != nil {
-			return err
+			return mki, err
 		}
 	}
 
 	n, err := io.Copy(sink, plainsource)
 	if err != nil {
-		return err
+		return mki, err
 	}
 
 	// TODO: Check header inline, and only warn if the footer
 	// doesn't match.
 	err = saltpack.CheckArmor62Frame(frame, saltpack.EncryptionArmorHeader, saltpack.EncryptionArmorFooter)
 	if err != nil {
-		return err
+		return mki, err
 	}
 
 	G.Log.Debug("Decrypt: read %d bytes", n)
```

Comment:
can SaltPackDecrypt have a GlobalContext as a parameter to avoid using capital G in new code?

That's what I did with SaltPackSign/Verify...